### ISSUE TITLE
Forbid to use ternary inside `if` clause

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#645]
# Rule request

## Thesis

One can write code like this:

```python
if one if some() else two:
    ...
```

## Reasoning

This is unreadable.
Use nested ifs instead.

**Comment by sobolevn** [comment]
Well, actually the rule is even stricter.
We should not be able to nest ternary anywhere except:
- `ast.Assign` and `ast.AnnAssign` as a value
- open to suggestions

**Comment by AlwxSin** [comment]
@sobolevn Why you forbid to use ternary but allow in some cases? It's kinda weird.

**Comment by sobolevn** [comment]
@AlwxSin yeap, since it is absolutely fine to write:

```python
some = x if cond() else y

print(x if cond() else y)  # is it?
```

But it is not ok to nest ternary inside a condition:

```python
x > (x if cond() else y)  # what?!
```

**Comment by sobolevn** [comment]
Ok, after some thinking, let's return to the original idea: we will only disallow ternaries inside:
- `if`
- `compare`
- `boolop` (>, +, *, ==, etc)
- `unary`: `-(x if cond() else y)`
